src/client_list.c

```diff
@@ -40,6 +41,11 @@ reg_list_item_t * get_global_node(void)
     return g_head;
 }
 
+pthread_mutex_t *get_global_client_mut(void)
```

Comment:
Instead of getter method for this mutex, we should add this lock and unlock logic inside the 
reg_list_item_t * get_global_node(void)getter function

Reply:
OK.  Then we should have release_global_node(), which will call unlock.